packages/playwright-core/src/server/browserContext.ts

```diff
@@ -137,6 +137,40 @@ export abstract class BrowserContext extends SdkObject {
       await mkdirIfNeeded(path.join(this._options.recordVideo.dir, 'dummy'));
   }
 
+  canResetForReuse(): boolean {
+    if (this._closedStatus !== 'open')
+      return false;
+    if (this.pages().length < 1)
+      return false;
+    return true;
+  }
+
+  async resetForReuse(metadata: CallMetadata) {
+    this.setDefaultNavigationTimeout(undefined);
+    this.setDefaultTimeout(undefined);
+
+    await this._cancelAllRoutesInFlight();
+    await this.setRequestInterceptor(undefined);
+
+    const [page, ...otherPages] = this.pages();
+    for (const page of otherPages)
+      await page.close(metadata);
+    // Unless I do this early, setting extra http headers below does not respond.
```

Comment:
Perhaps the same is true for `setRequestInterceptor`?

Reply:
Done!